common/datavalues/src/columns/boolean/mod.rs

```diff
@@ -103,17 +103,37 @@ impl Column for BooleanColumn {
     }
 
     fn filter(&self, filter: &BooleanColumn) -> ColumnRef {
-        if filter.values().null_count() == 0 {
+        let selected = filter.values().len() - filter.values().null_count();
+        if selected == self.len() {
             return Arc::new(self.clone());
         }
-        let iter = self
-            .values()
-            .iter()
-            .zip(filter.values().iter())
-            .filter(|(_, b)| *b)
-            .map(|(a, _)| a);
+        let mut bitmap = MutableBitmap::with_capacity(selected);
+        let mut chunks = self.values().chunks::<u64>();
```

Comment:
I think we should follow the way of `primitive::filter`.

Iterator of `filter.values().chunks::<u64>` may cause too much  u64 merges.